internal/api/handler.go

```diff
@@ -341,8 +346,17 @@ func (h *Handler) handleTokenExchange(c *OAuthAppContext, w http.ResponseWriter,
 
 	h.metricsCollector.IncGeneratedToken(c.OAuthApplication.Identifier)
 
+	expiry := token.Extra("expires_in").(float64)
+	tokenResponse := &TokenResponse{
+		ExpiresIn: expiry,
+	}
+	tokenResponse.AccessToken = token.AccessToken
+	tokenResponse.RefreshToken = token.RefreshToken
+	tokenResponse.TokenType = token.TokenType
+	tokenResponse.Expiry = token.Expiry
```

Comment:
```suggestion
	tokenResponse := &TokenResponse{
		ExpiresIn: expiry,
		AccessToken: token.AccessToken
	        RefreshToken: token.RefreshToken
	        TokenType: token.TokenType
	        Expiry: token.Expiry
	}
```

Reply:
I tried this but it doesn't work. We cannot directly access fields of a nested struct while creating a struct literal.

Reply:
I didn't know this, thanks